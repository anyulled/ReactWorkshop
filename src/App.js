import React, {Component} from "react";
import {Grid, PageHeader, Row, Col, Glyphicon} from "react-bootstrap";
import WeatherForecast from "./WeatherForecast";

class App extends Component {
    constructor() {
        super();
        this.state = {
            city: ""
        };
        this.modifyCity = this.modifyCity.bind(this);
    }

    modifyCity(city) {
        this.setState({
            city
        });
    }

    render() {
        return (
            <div>
                <PageHeader><Glyphicon glyph="grain"/> {this.state.city || "Weather App"  }</PageHeader>
                <Grid>
                    <Row>
                        <Col sm={12}>
                            <WeatherForecast modifyCity={this.modifyCity}/>
                        </Col>
                    </Row>
                </Grid>
            </div>
        );
    }
}

export default App;
