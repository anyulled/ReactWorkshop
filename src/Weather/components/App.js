import React, {Component} from "react";
import {Grid, Row, Col} from "react-bootstrap";
import WeatherContainer from "../container/WeatherContainer";
import HeaderContainer from "../container/HeaderContainer";
import StatisticsContainer from "../container/StatisticsContainer";

class App extends Component {
    render() {
        return (
            <div>
                <HeaderContainer/>
                <Grid>
                    <Row>
                        <Col sm={12}>
                            <WeatherContainer/>
                        </Col>
                    </Row>
                    <Row>
                        <Col sm={6} smOffset={3}>
                            <StatisticsContainer/>
                        </Col>
                    </Row>
                </Grid>
            </div>
        );
    }
}

export default App;
