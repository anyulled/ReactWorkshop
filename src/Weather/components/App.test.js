import React from "react";
import App from "./App";
import {shallow} from "enzyme";

describe("App", () => {
    it("Renders properly", () => {
        const wrapper = shallow(<App/>);
        expect(wrapper).toMatchSnapshot();
    });

    it("displays both weather and statistics container", () => {
        const wrapper = shallow(<App/>);
        expect(wrapper.find("Connect(Header)").exists()).toBe(true);
        expect(wrapper.find("Connect(WeatherForecast)").exists()).toBe(true);
        expect(wrapper.find("Connect(Statistics)").exists()).toBe(true);
    });
});