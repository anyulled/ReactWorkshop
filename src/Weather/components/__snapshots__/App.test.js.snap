// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <Grid
          bsClass="container"
          componentClass="div"
          fluid={false}
        >
          <Row
            bsClass="row"
            componentClass="div"
          >
            <Col
              bsClass="col"
              componentClass="div"
              sm={12}
            >
              <Connect(WeatherForecast) />
            </Col>
          </Row>
          <Row
            bsClass="row"
            componentClass="div"
          >
            <Col
              bsClass="col"
              componentClass="div"
              sm={6}
              smOffset={3}
            >
              <Connect(Statistics) />
            </Col>
          </Row>
        </Grid>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "bsClass": "container",
          "children": Array [
            <Row
              bsClass="row"
              componentClass="div"
            >
              <Col
                bsClass="col"
                componentClass="div"
                sm={12}
              >
                <Connect(WeatherForecast) />
              </Col>
            </Row>,
            <Row
              bsClass="row"
              componentClass="div"
            >
              <Col
                bsClass="col"
                componentClass="div"
                sm={6}
                smOffset={3}
              >
                <Connect(Statistics) />
              </Col>
            </Row>,
          ],
          "componentClass": "div",
          "fluid": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "bsClass": "row",
              "children": <Col
                bsClass="col"
                componentClass="div"
                sm={12}
              >
                <Connect(WeatherForecast) />
              </Col>,
              "componentClass": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": <Connect(WeatherForecast) />,
                "componentClass": "div",
                "sm": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "bsClass": "row",
              "children": <Col
                bsClass="col"
                componentClass="div"
                sm={6}
                smOffset={3}
              >
                <Connect(Statistics) />
              </Col>,
              "componentClass": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": <Connect(Statistics) />,
                "componentClass": "div",
                "sm": 6,
                "smOffset": 3,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Grid
            bsClass="container"
            componentClass="div"
            fluid={false}
          >
            <Row
              bsClass="row"
              componentClass="div"
            >
              <Col
                bsClass="col"
                componentClass="div"
                sm={12}
              >
                <Connect(WeatherForecast) />
              </Col>
            </Row>
            <Row
              bsClass="row"
              componentClass="div"
            >
              <Col
                bsClass="col"
                componentClass="div"
                sm={6}
                smOffset={3}
              >
                <Connect(Statistics) />
              </Col>
            </Row>
          </Grid>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "bsClass": "container",
            "children": Array [
              <Row
                bsClass="row"
                componentClass="div"
              >
                <Col
                  bsClass="col"
                  componentClass="div"
                  sm={12}
                >
                  <Connect(WeatherForecast) />
                </Col>
              </Row>,
              <Row
                bsClass="row"
                componentClass="div"
              >
                <Col
                  bsClass="col"
                  componentClass="div"
                  sm={6}
                  smOffset={3}
                >
                  <Connect(Statistics) />
                </Col>
              </Row>,
            ],
            "componentClass": "div",
            "fluid": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "bsClass": "row",
                "children": <Col
                  bsClass="col"
                  componentClass="div"
                  sm={12}
                >
                  <Connect(WeatherForecast) />
                </Col>,
                "componentClass": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "col",
                  "children": <Connect(WeatherForecast) />,
                  "componentClass": "div",
                  "sm": 12,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "bsClass": "row",
                "children": <Col
                  bsClass="col"
                  componentClass="div"
                  sm={6}
                  smOffset={3}
                >
                  <Connect(Statistics) />
                </Col>,
                "componentClass": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "col",
                  "children": <Connect(Statistics) />,
                  "componentClass": "div",
                  "sm": 6,
                  "smOffset": 3,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
